test/EFCore.Analyzers.Tests/Analyzers/InternalUsageDiagnosticAnalyzerTests.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.EntityFrameworkCore.Analyzers;
+
+using VerifyCS = CSharpAnalyzerVerifier<InternalUsageDiagnosticAnalyzer>;
+
+public class InternalUsageDiagnosticAnalyzerTests
+{
+    [ConditionalFact]
+    public Task Invocation_on_type_in_internal_namespace()
+        => VerifySingleInternalUsageAsync("""
+            using System;
+            using Microsoft.EntityFrameworkCore.Internal;
+
+            class C
+            {
+                void M()
+                {
+                    var x = typeof(object).GetMethod(nameof(object.ToString), Type.EmptyTypes).{|#0:DisplayName|}();
+                }
+            }
+            """, "Microsoft.EntityFrameworkCore.Internal.MethodInfoExtensions");
```

Comment:
nit: my preference is to align the code to the beginning of the line, like we do e.g. with SQL assertions (some code samples here bypass the 120 char limit).

```suggestion
        => VerifySingleInternalUsageAsync("""
using System;
using Microsoft.EntityFrameworkCore.Internal;

class C
{
    void M()
    {
        var x = typeof(object).GetMethod(nameof(object.ToString), Type.EmptyTypes).{|#0:DisplayName|}();
    }
}
""", "Microsoft.EntityFrameworkCore.Internal.MethodInfoExtensions");
```